Update SSH Agent FAQ for Windows

Okeanos opened the pull request:
Previously Windows' own OpenSSH option was not mentioned, even though KeePassXC supports it.
Additionally, to better differentiate between the two major platforms (Unix vs. Window) and their differences they now each have a separate section.

Some additional comments:

I removed the line following line from the `Why is Pageant refusing my keys?`-question:
> There doesn't seem to be any alternative to Pageant for Windows that supports both of them.

Because from a cursorily glance at the [Win32 OpenSSH issues](https://github.com/PowerShell/Win32-OpenSSH/issues?q=is%3Aissue+askpass) there may be options to correctly provide and configure `SSH_ASKPASS` when using Windows' own OpenSSH binaries. I haven't verified that.

There is "a lot" of duplication between the two sections – yeah, that's true. It might be useful to split that into a generalised third section. Thoughts?

It might be worth a shot to add a question/answer pair outlining the "use KeePassXC together with Git for Windows" workaround/solution I referred to in [keepassxc#4681](https://github.com/keepassxreboot/keepassxc/issues/4681).

Assuming any changes have to be made to get this merged … I assume you want all of this in a single commit?

As an aside, unrelated to this pull request: My IDE and I noticed some trailing whitespaces throughout the file and generally inconsistent formatting (e.g. `<br>` vs `<br />`)  and some technical problems (such as missing references e.g. to `#faq-string-fields` or wrongly nested HTML elements `<ul>` inside of `<ul>` instead of `<li>`). Would you be interested in a separate pull request that fixes these things?

hifi commented:
@phoerious @droidmonkey should we just remove the outdated FAQ entries (not all) and point people to the docs? I don't think maintaining the FAQ as documentation like it was before is good use of anyone's time.

@Okeanos is the actual documentation at https://keepassxc.org/docs/KeePassXC_UserGuide.html#_ssh_agent more up to date and would you be interested in working on that? The doc sources are part of the main KeePassXC repository.

phoerious commented:
I would keep some FAQ, but if things can be explained better in the docs, I'm all for it.

Okeanos commented:
The actual documentation does indeed look better and I'd be willing to see whether I can still improve it.

Reducing the FAQ and at the same time fixing some issues (like pointing out OpenSSH is supported on Windows) sounds like a good idea.

droidmonkey commented:
I dislike the FAQ for several reasons. That page is ridiculously large and impossible to navigate. The information tends to get stale quickly. Anything that is truly "documentation" should be put in our user guide instead. FAQ should be errata, not fundamental usage tips.

Okeanos commented:
I now reverted most of my original changes and instead opted to enrich the existing SSH Agent information with minimal additions concerning the Windows support.

If there are additional changes required before this can be merged I'd be happy to do that. I would like to see the FAQ changes merged because until a better version exists I would at least like to get rid of some inaccuracies for now.

Concerning the full rewrite/change of the FAQ and updates to the real user documentation: I would personally prefer to do that in a separate pull request; especially the former currently appears to require more discussion?

droidmonkey commented:
Documentation is edited on the keepassxc repo. https://github.com/keepassxreboot/keepassxc/tree/develop/docs

Okeanos commented:
I updated the pull request and added a single point concerning interaction with Git For Windows – if there particular changes I have to make before this can be merged I'd be happy to do so.

Okeanos commented:
@hifi I just noticed that this is still open and I am wondering whether there are any additional changes required or different plans altogether for the FAQ (that I might be able contribute to)? I'd appreciate some feedback, even if it's just closing/declining the pull request.

Okeanos commented:
@droidmonkey @hifi I updated the pull request to resolve the conflicts with the current master. If there's anything else I need to change for this to get merged, please let me know.

hifi commented:
Ah, sorry. I'll read these through and merge if nothing stands out. Thanks!

hifi commented on docs/index.html:
```diff
@@ -489,15 +501,18 @@ <h5>SSH Agent integration</h5>
         <dd>
-            Pageant does not support confirm-on-use or automatic removal of key after a timeout. There doesn't seem to be
-            any alternative to Pageant for Windows that supports both of them.
+            Pageant does not support confirm-on-use or automatic removal of key after a timeout.
```
Has this changed? AFAIK OpenSSH for Windows does not support confirm-on-use or automatic removal after a timeout.

Okeanos replied:
That's a fair question, I'll verify that tomorrow!

hifi commented on docs/index.html:
```diff
@@ -489,15 +501,18 @@ <h5>SSH Agent integration</h5>
 
         <dt id="faq-ssh-agent-pageant"><a href="#faq-ssh-agent-pageant">Why is Pageant refusing my keys?</a></dt>
```
This should probably be expanded to Pageant and OpenSSH (both versions?) if what I noted below is still true.

Okeanos replied:
Will check – I didn't put anything in here because I simply didn't verify that behaviour at the time and the consensus was not to expand the FAQs too much.

Okeanos replied:
I added a sentence explaining that Windows OpenSSH is not compatible with confirm-on-use and timeouts after verifying the behaviour locally.

hifi commented on docs/index.html:
```diff
@@ -523,6 +538,17 @@ <h5>SSH Agent integration</h5>
             your public key file. The SSH Agent will use the provided information to select the correct private key.
         </dd>
+
+        <dt id="faq-ssh-agent-git-bash"><a href="#faq-ssh-agent-git-bash">How do I use KeePassXC SSH Agent integration with Git (Bash) on Windows?</a></dt>
```
This should be expanded to include how to change the `ssh.exe` used by Git to Windows 10 OpenSSH as that's the most native way.

Okeanos replied:
Ah, it's a little implicit I grant you that! There's three ways to achieve that: have Git for Windows 10+ use the Windows supplied version when prompted during installation of Git, modify the Git Bash `$PATH` by prepending the Windows OpenSSH Path – that makes Git for Windows prefer it to its own (if installed) – or set the `GIT_SSH_COMMAND` variable. The last option will _only_ work for Git, however, no other binary in the Git Bash will sadly pick it up leading to some weird edge cases. Shall I include this in the FAQ?

Okeanos replied:
I added an additional sentence explaining `$PATH` and `$GIT_SSH_COMMAND` options.

hifi reviewed (changes requested):
I read it through and noted some things that could be improved.

Sorry for forgetting this PR completely, it's not against the main repo so it doesn't pop up in my views that easily. :bow:

droidmonkey commented:
@hifi good to go?

hifi reviewed (approved):
LGTM, thanks a lot for sticking through!
